src/apps/replication/lib/replica_2pc.cpp

```diff
@@ -56,7 +56,7 @@ void replica::on_client_write(int code, dsn_message_t request)
         return;
     }
 
-    auto mu = _primary_states.write_queue.add_work(code, request, this);
+    auto mu = _primary_states.write_queue.add_work(code, request, this, _options->batch_write_disabled);
```

Comment:
batch-write-disabled can be achieved by another simpler and more general approach